### ISSUE TITLE
Input file for Megalodon mapping validation 

### ISSUE THREAD

**Issue by dipannita-g** [nanoporetech/megalodon#188]
Hi! 

Just wanted to know what exact input file is needed for the mapping validation? 
My Megalodon result output folder has **mod_mappings.bam**  file but the `megalodon_extras validate results` says mappings not found and only gives me the ROC curves and base score distribution plot. This is the output argument for my Megalodon run:
`--outputs mod_basecalls mod_mappings per_read_mods mods`
Do I need to specify mapping argument --mappings-format while running Megalodon? If yes, which format- bam/cram/sam? 

Thanks
Dipannita

**Comment by marcus1487** [comment]
TL;DR the `--outputs mappings` setting will produce the files needed to produce the mapping validation files. Specifically the mapping validation uses the `mappings.summary.txt` file, which includes summary metrics about the mappings in `mappings.[bam|sam|cram]`.

For some more detail, the `mod_mappings.bam` file does not contain traditional "mappings". This file swaps the basecalls/called sequence for the reference and will always show exact matches for every read. This is because megalodon makes more accurate modified base calls by including information from the reference than only using the original basecalls (this is really the core functionality of megalodon). We are working to close this gap and make basecall anchored calls more accurate.

What this means is that the basecall accuracy mapping statistics are not included in the mod_mappings output. The mod_mappings are useful for performing analysis on per-read modified base calls made against all reference bases. Note that the mod_basecalls mapped to the reference will miss some reference bases due to indels and SNPs in the mappings.

**Comment by dipannita-g** [comment]
Thank you Marcus for the details.
